cpp/src/gandiva/precompiled/time.cc

```diff
@@ -828,4 +841,22 @@ gdv_int64 castBIGINT_daytimeinterval(gdv_day_time_interval in) {
          extractDay_daytimeinterval(in) * MILLIS_IN_DAY;
 }
 
+// Convert the seconds since epoch argument to timestamp
+#define TO_TIMESTAMP(TYPE)                                      \
+  FORCE_INLINE                                                  \
+  gdv_timestamp to_timestamp##_##TYPE(gdv_##TYPE seconds) {     \
+    return static_cast<gdv_timestamp>(seconds) * MILLIS_IN_SEC; \
+  }
+
+NUMERIC_TYPES(TO_TIMESTAMP)
+
+// Convert the seconds since epoch argument to time
+#define TO_TIME(TYPE)                                        \
+  FORCE_INLINE                                               \
+  gdv_time32 to_time##_##TYPE(gdv_##TYPE seconds) {          \
+    return static_cast<gdv_time32>(seconds) * MILLIS_IN_SEC; \
```

Comment:
This is incorrect. Time refers to the millis since midnight. You are returning the timestamp. You need to get the time of the day of this timestamp

Reply:
@projjal I'm still a littlle bit confused with your comment. Take a look at the following code:
```cpp
// Convert the seconds since epoch argument to time
#define TO_TIME(TYPE)                                  \
  FORCE_INLINE                                         \
  gdv_timestamp to_time##_##TYPE(gdv_##TYPE seconds) { \
    EpochTimePoint tp(seconds * MILLIS_IN_SEC);        \
    return tp.TmSec() * MILLIS_IN_SEC;                 \
  }
```
Some questions:
1 - Is the operation now correct?
2 - Should the return type be a timestamp?

Reply:
It should be something like 
```
EpochTimePoint tp(seconds * MILLIS_IN_SEC);
return tp.TimeOfDay().to_duration().to_millis()
```

Reply:
Now I think I did it correctly :)

Solved!